api/src/main/java/org/apache/cloudstack/api/command/user/backup/RestoreVolumeFromBackupAndAttachToVMCmd.java

```diff
@@ -73,6 +74,12 @@ public class RestoreVolumeFromBackupAndAttachToVMCmd extends BaseAsyncCmd {
             description = "id of the VM where to attach the restored volume")
     private Long vmId;
 
+    @Parameter(name = ApiConstants.START_VM,
+            type = CommandType.BOOLEAN,
+            required = false,
+            description = "start VM after the restore is finished")
```

Comment:
```suggestion
    @Parameter(name = ApiConstants.START_VM,
            type = CommandType.BOOLEAN,
            required = false,
            description = "start VM after the restore is finished",
            since = "4.19.0.0")
```

Reply:
this will power on the vm from veeam, looks like a out-of-band operation (the power state, host_id, etc are not managed by cloudstack).

@SadiJr 
I think we should avoid it. cc @DaanHoogland

Reply:
@weizhouapache when using the option to power on VM from Veeam, it will only start the VM in the last host the VM was allocated to. Also, there is a process in ACS that updates the VM state from time to time, according to what is collected from vCenter. Additionally, the HA process in VMWare is also not managed by ACS. Having said all that, I believe that if Veeam has this option, we should leave it up to the user to decide, not us.

Reply:
@SadiJr 
You can do lots of operations on vcenter or veeam, but cloudstack does not support OOB operations perfectly.

We should avoid the OOB operations as much as possible, unless it is really neccessary.
I think it is unnecessary to add the option. If you insist to support, I would rather starting the vm by invoking cloudstack methods/API (as cloudstack will check the proper host by available resources, affinity groups, and apply network rules, etc), than starting the vm via veeam. 

Let me give you an example, if you stop a vm, and the network is shutdown (VR is stopped), when you start the vm in vcenter or veeam, the vm has no internet connection as the VR is still stopped.

Reply:
@weizhouapache I have removed this feature from this PR as it is not in the core of it. In the future I might open a PR to add it separately and we can discuss it further on that PR.